axelor-production/src/main/java/com/axelor/apps/production/web/AppProductionController.java

```diff
@@ -32,4 +34,17 @@ public void generateProductionConfigurations(ActionRequest request, ActionRespon
 
     response.setReload(true);
   }
+
+  public void checkIfOutsourcingDisabled(ActionRequest request, ActionResponse response) {
+    AppProduction app = request.getContext().asType(AppProduction.class);
+    if (app.getId() == null) {
+      return;
+    }
+    boolean outsourcing =
+        Beans.get(AppProductionRepository.class).find(app.getId()).getManageOutsourcing();
+    boolean outsourcingContxt = app.getManageOutsourcing();
+    if (outsourcing && !outsourcingContxt) {
+      Beans.get(AppProductionService.class).updatePartnerSubcontractTag();
+    }
```

Comment:
All exception must be traced inside a controller method, please enclose everything in a try {} catch {} or use the new annotation `@HandleExceptionResponse`